### PR TITLE
UX: Added e2e for list management and removed feature flag

### PR DESCRIPTION
**Pull request by NidhiKJha**
This PR is to:

- Remove the `NETWORK_ACCOUNT_DND` flag
- Add e2e tests for networks and accounts. NOTE: for drag and drop, it doesn't have an e2e.
- Added banners for account pinning and network drag and drop.

## **Related issues**

Fixes: [issues/1845](https://github.com/MetaMask/MetaMask-planning/issues/1845)

## **Manual testing steps**

1. Go to extension
2. Open the network list menu, try dragging and dropping different non-test networks
3. For the first time, the banner will show up to demo drag and drop. Once you click the close banner button, it shouldn't show up
4. Open the account-list menu, try pinning and hiding accounts
5. For the first time, the banner will show up to demo account pinning. Once you click the close banner button, it shouldn't show up

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-extension/assets/39872794/5d252605-190f-488e-b2fc-c91c65b29d35

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/account-details.spec.js [hunk 1]
```diff
@@ -205,4 +205,134 @@ describe('Show account details', function () {
...
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        await driver.clickElement('[data-testid="account-menu-icon"]');
+        await driver.clickElement(
+          '[data-testid="account-list-item-menu-button"]',
+        );
+        await driver.clickElement('[data-testid="account-list-menu-pin"]');
```
Comment (DDDDDanica) [inline test/e2e/tests/account-details.spec.js:222]: Can we extract them as a small helper function, namely `pingAccountItem` ?

**Review by darkwing — COMMENTED** [review]
Wow, this is great!

**Comment by darkwing** [comment]
Unfortunately we need to update the PR template here with the "Before" and "After" headings for this to pass. :/

**Comment by HowardBraham** [comment]
This branch consistently fails on "Chain Interactions should add the Ganache test chain and not switch the network"

**Comment by NidhiKJha** [comment]
> This branch consistently fails on "Chain Interactions should add the Ganache test chain and not switch the network"

Yeah, the test fails because it's unable to detect the network. However within the automated test, if I close the network menu and then open it, I can see the network. Also, the console gets updated till this stage in my local. 😞

File: test/e2e/tests/account-details.spec.js (hunk 1 above)
Comment (NidhiKJha) [reply test/e2e/tests/account-details.spec.js:222]: Hmmm...We can but for two cases we have this same function and then it's hiding and for some cases we don't have to define pinnedIcon again. So, I thought we can keep it like this. Wdyt?

**Review by DDDDDanica — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**